### PR TITLE
Stop run.d if any tools fail to build

### PR DESCRIPTION
**Pull request by marler8997**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: test/run.d [hunk 1]
```diff
@@ -166,9 +167,15 @@ void ensureToolsExists(const TestTool[] tools ...)
...
 
             writefln("Executing: %-(%s %)", command);
-            spawnProcess(command).wait;
+            if (spawnProcess(command).wait)
+                failCount++;
         }
     }
+    if (failCount)
+    {
+        stderr.writefln("failed to build %s test tool(s)", failCount);
```
Comment (jacob-carlborg) [inline test/run.d]: It would be better to report the names of the tools that failed than the count.

Comment (marler8997) [reply test/run.d]: That information should already be printed, we actually would probably be fine not printing anything here.  What do you think?

Comment (jacob-carlborg) [reply test/run.d]: If the information is already printed I'm fine with this. Perhaps even unnecessary to print the count.

Comment (marler8997) [reply test/run.d]: ok, no longer printing count, and making sure to print which tool(s) failed

**Review by jacob-carlborg — APPROVED** [review]

**Comment by marler8997** [comment]
Updated `failCount` to be `shared`, and now I used `atomicOp` to modify it since the variable is written to from multiple threads.

**Pull request merged**